src/effects/Dust.tsx

```diff
@@ -0,0 +1,69 @@
+import { Vector3, Matrix4, Object3D, Quaternion, MathUtils } from 'three'
+import { useRef } from 'react'
+import { useFrame } from '@react-three/fiber'
+import type { InstancedMesh } from 'three'
+import type { MutableRefObject } from 'react'
+import type { CannonApi } from '../store'
+import { getState, mutation, useStore } from '../store'
+
+const v = new Vector3()
+const m = new Matrix4()
+const o = new Object3D()
+const q = new Quaternion()
+
+interface DustProps {
+  count?: number
+  opacity?: number
+  size?: number
+}
+
+export function Dust({ count = 200, opacity = 0.1, size = 1 }: DustProps): JSX.Element {
+  const { wheels } = useStore((state) => state.raycast)
+  const ref = useRef<InstancedMesh>(null!)
+
+  let i = 0
+  let index = 0
+  let time = 0
+  let intensity = 0
+  let controls: ReturnType<typeof getState>['controls']
```

Comment:
`ReturnType<typeof getState>['controls']` -> `Store['controls']`
I don't see the need to involve `getState`. You can directly get it from the interface - We even have a specific Controls interface you could use.